src/udev/udev-builtin-net_id.c

```diff
@@ -349,6 +352,50 @@ static int names_vio(struct udev_device *dev, struct netnames *names) {
         return 0;
 }
 
+#define _PLATFORM_TEST "/sys/devices/platform/vvvvPPPP"
+#define _PLATFORM_PATTERN4 "/sys/devices/platform/%4s%4x:%2x/net/eth%u"
+#define _PLATFORM_PATTERN3 "/sys/devices/platform/%3s%4x:%2x/net/eth%u"
+
+static int names_platform(struct udev_device *dev, struct netnames *names, bool test) {
+        struct udev_device *parent;
+        char vendor[5];
+        char shortid;
+        unsigned model, instance, ethid;
+        const char *syspath, *pattern;
+
+        /* check if our direct parent is a platform device with no other bus in-between */
+        parent = udev_device_get_parent(dev);
+        if (!parent)
+                return -ENOENT;
+
+        if (!streq_ptr("platform", udev_device_get_subsystem(parent)))
+                 return -ENOENT;
+
+        syspath = udev_device_get_syspath(dev);
+
+        /* Vendor ID inside ACPI id can be either 3 or 4 */
+        if (syspath[sizeof _PLATFORM_TEST - 1] == ':')
+                pattern = _PLATFORM_PATTERN4;
+        else
+                pattern = _PLATFORM_PATTERN3;
+
+        /* Platform devices are named after ACPI table match, and instance id
+         * eg. "/sys/devices/platform/HISI00C2:00");
+         * The Vendor (3 or 4 char), followed by hexdecimal model number : instance id.
+         */
+        if (sscanf(syspath, pattern, vendor, &model, &instance, &ethid) != 4)
+                 return -EINVAL;
+
+        if (streq(vendor, "HISI"))
+                shortid = 'h';
+        else
+                return -ENOENT;
```

Comment:
The vendor id is long 4 characters and HISI stands for Hisilicon and is too long for me imho, ditto hexidecimal 00c2 is too long as well, hense I strip the leading zeros. Hence instead of enahisimc2p1 I shorten the vendor id to just letter 'h'.

I guess we could lowercase the vendor ID, and then don't need a separator for the model and thus generate names without harcoding a mapping for ACPI allocated vendor IDs that are in use. E.g.:
 * enahisic2i0
 * enahisic2i1
 * enahisic2i2
 * enahisic2i3